NineChronicles.Headless/GraphTypes/StateQuery.cs

```diff
@@ -91,9 +92,80 @@ public StateQuery()
+                    var round = context.GetArgument<int>("round");
+                    var sheets = context.Source.GetSheets(sheetTypes: new[]
+                    {
+                        typeof(ArenaSheet)
+                    });
+                    var arenaSheet = sheets.FirstOrDefault().Value.sheet as ArenaSheet;
+                    if (arenaSheet == null || !arenaSheet.TryGetValue(championshipId, out var arenaRow))
+                    {
+                        throw new SheetRowNotFoundException(nameof(ArenaSheet),
+                            $"championship Id : {championshipId}");
+                    }
+                    if (!arenaRow.TryGetRound(round, out var roundData))
+                    {
+                        throw new RoundNotFoundException(
+                            $"[{nameof(BattleArena)}] ChampionshipId({arenaRow.ChampionshipId}) - round({round})");
+                    }
+
+                    var arenaParticipantsAdr =
+                        ArenaParticipants.DeriveAddress(roundData.ChampionshipId, roundData.Round);
+                    if (!context.Source.TryGetArenaParticipants(arenaParticipantsAdr, out var arenaParticipants))
+                    {
+                        throw new ArenaParticipantsNotFoundException(
+                            $"[{nameof(BattleArena)}] ChampionshipId({roundData.ChampionshipId}) - round({roundData.Round})");
+                    }
+                    var championshipInfo = new ChampionshipArenaStateType();
+                    List<ChampionArenaInfo> arenaInformations = new List<ChampionArenaInfo>();
+                    //championshipInfo.
+                    foreach (var participant in arenaParticipants.AvatarAddresses)
+                    {
+                        var arenaInformationAdr =
+                            ArenaInformation.DeriveAddress(participant, roundData.ChampionshipId, roundData.Round);
+                        if (!context.Source.TryGetArenaInformation(arenaInformationAdr, out var arenaInformation))
+                        {
+                            continue;
+                        }
+                        var arenaScoreAdr =
+                                ArenaScore.DeriveAddress(participant, roundData.ChampionshipId, roundData.Round);
+                        if (!context.Source.TryGetArenaScore(arenaScoreAdr, out var arenaScore))
+                        {
+                            continue;
+                        }
+
+                        var arenaInfo = new ChampionArenaInfo();
+                        arenaInfo.AvatarAddress = participant;
+                        arenaInfo.Win = arenaInformation.Win;
+                        arenaInfo.Ticket = arenaInformation.Ticket;
+                        arenaInfo.Lose = arenaInformation.Lose;
+                        arenaInfo.Score = arenaScore.Score;
+                        arenaInformations.Add(arenaInfo);
```

Comment:
If you want to sorting this `arenaInformations` same with client, and check [this function][1].

- Ranking is determined by score.
- For duplicate ranks, the same rank is displayed.
   - In this case, the same rank is a rank that is separated by the number of duplicates from the previous rank.
- Duplicate Rank The next rank is displayed as a rank separated by the number of duplicates.

e.g., You can check what we expect with [Data source][2] and [Unit test][3]

And we will move this ranking function from [9c][4] to [lib9c][5]


[1]: https://github.com/planetarium/NineChronicles/blob/main/nekoyume/Assets/_Scripts/State/RxProps.Arena.cs#L452-L526
[2]: https://github.com/planetarium/NineChronicles/blob/main/nekoyume/Assets/Tests/EditMode/State/RxPropsTest.cs#L10-L38
[3]: https://github.com/planetarium/NineChronicles/blob/main/nekoyume/Assets/Tests/EditMode/State/RxPropsTest.cs#L10-L38
[4]: https://github.com/planetarium/NineChronicles
[5]: https://github.com/planetarium/lib9c